mkosi/qemu.py

```diff
@@ -788,6 +788,22 @@ def finalize_state(config: Config, cid: int) -> Iterator[None]:
             p.unlink(missing_ok=True)
 
 
+def finalize_kernel_command_line(
+    config: Config,
+    kernel: Optional[Path],
+    firmware: QemuFirmware,
+) -> list[str]:
+    if kernel and (
+        KernelType.identify(config, kernel) != KernelType.uki
+        or not config.architecture.supports_smbios(firmware)
+    ):
+        cmdline = [*config.kernel_command_line]
```

Comment:
I actually wondered why this is done. It obviously is required for the eventual call to qemu but prepending it this early leads to a different behavior when directly booting a kernel vs normal boot if the normal cmdline contains a `root=` (as in one case it will be picked up by the check below and in the other case it will not)